extensions/functions_comparison.yaml

```diff
@@ -73,4 +73,12 @@ scalar_functions:
           - value: any1
         return: BOOLEAN
         nullability: DECLARED_OUTPUT
+  -
+    name: "is_nan"
+    description: Whether a value is not a number.
+    impls:
+      - args:
+          - value: any1
+        return: BOOLEAN
+        nullability: DECLARED_OUTPUT
```

Comment:
Why is this declared output? Shouldn't nan be null if the input is null?

Reply:
I don't have strong feelings (null is not nan) but Arrow seems to agree with @jacques-n:

```
>>> pc.is_nan(None)
<pyarrow.BooleanScalar: None>
```